java-connectors/kafka-connect-common/src/main/java/io/lenses/streamreactor/common/util/JarManifest.java

```diff
@@ -69,11 +73,13 @@ public JarManifest(JarFile jarFile) {
     Manifest manifest;
     try {
       Optional<JarFile> jarFileOptional = of(jarFile);
-      manifest = jarFileOptional.get().getManifest();
+      if (jarFileOptional.isPresent()) {
+        manifest = jarFileOptional.get().getManifest();
+        extractMainAttributes(manifest.getMainAttributes());
+      }
     } catch (NullPointerException | IOException e) {
       throw new ConnectorStartupException(e);
     }
-    extractMainAttributes(manifest.getMainAttributes());
   }
 
   private void extractMainAttributes(Attributes mainAttributes) {
```

Comment:
Food for thought : 🍕 

When a function returns void, it typically indicates that the function performs side effects (such as modifying mutable state, printing to the console, or writing to a file) without producing any meaningful result. Such functions are not referentially transparent because their effects are observable and can change the behaviour of subsequent function calls.

instead of this returning void, it would be good for referential transparancy if this constructed the map immutably and returned the new map, to be set into the class field (which could also be an Optional<>, empty when uninitialised.

Reply:
Good one! I used this idea for `extractMainAttributes` and `buildManifestString` function as well!